src/_pytest/stepwise.py

```diff
@@ -71,30 +66,24 @@ def pytest_collection_modifyitems(
         self, config: Config, items: List[nodes.Item]
     ) -> None:
         if not self.lastfailed:
-            self.report_status = NO_PREVIOUSLY_FAILED_WONT_SKIP
+            self.report_status = "no previously failed tests, not skipping."
             return
-        already_passed = []
-        found = False
 
-        # Make a list of all tests that have been run before the last failing one.
-        for item in items:
+        # check all item nodes until we find a match on last failed
+        failed_index = None
+        for index, item in enumerate(items):
             if item.nodeid == self.lastfailed:
-                found = True
+                failed_index = index
                 break
-            else:
-                already_passed.append(item)
+
         # If the previously failed test was not found among the test items,
         # do not skip any tests.
-        if not found:
-            self.report_status = PREVIOUSLY_FAILED_TEST_NOT_FOUND
-            already_passed = []
+        if not failed_index:
+            self.report_status = "previously failed test not found, not skipping."
         else:
-            self.report_status = SKIPPING_PASSED_ITEMS.format(len(already_passed))
-
-        for item in already_passed:
-            items.remove(item)
-
-        config.hook.pytest_deselected(items=already_passed)
+            self.report_status = f"skipping {failed_index} already passed items."
+            items[:] = items[failed_index:]
+            config.hook.pytest_deselected(items=items[:failed_index])
```

Comment:
This is incorrect, `items` at this point only contains the failed item and the other items after it (see my suggestion).

Reply:
Hmmm that no tests caught this is troubling actually, most likely because this only affects the output summary.

Reply:
whoops my bad, will also add coverage there - yes very troubling :)

Reply:
@nicoddemus I (hope) my latest commit can cover this scenario, tho if you could have a look that would be appreciated.  I plan as a follow up (subsequent ticket/PR) to move away from `testdir` here and use `pytester` instead.

Reply:
@symonk If you want to test this, you can add a `pytest_deselected` hookimpl in a `conftest.py` file in the test, and check that it receives what is expected.